tests/webTables.spec.ts

```diff
@@ -0,0 +1,177 @@
+
+            if (cityValue == 'Madison') {
+                petList.push(petsValue)
+            }
+        }
+        expect(petList).toEqual(expectedList)
+
+    })
+})
+
+test('Validate specialty update', async ({ page }) => {
+    await page.goto("/");
+    const veterinariansMenu = page.getByText(" Veterinarians")
+    await veterinariansMenu.click();
+    const veterinariansMenuAll = page.locator(".dropdown-menu").getByText(" All")
+    await veterinariansMenuAll.click();
+    const rafaelSpecialty = await page.getByRole('row', { name: " Rafael Ortega " }).locator('td').nth(1)
+    expect(await rafaelSpecialty.textContent()).toContain('surgery')
+
+    await page.getByRole('link', { name: 'Specialties' }).click()
+    await expect(page.locator("h2")).toHaveText("Specialties");
+
+    const rowSurgery = page.locator('tr', { has: page.locator('[id="1"]') })
+    await rowSurgery.getByRole("button", { name: "Edit" }).click();
+    await expect(page.locator('h2')).toContainText('Edit Specialty')
+    const editSpecialtyInput = page.locator('#name')
+    await editSpecialtyInput.click()
+    await editSpecialtyInput.clear()
+    await editSpecialtyInput.fill('dermatology')
+    await page.getByRole('button', { name: "Update" }).click()
+    await expect(rowSurgery.locator('td input').first()).toHaveValue('dermatology')
+
+    await veterinariansMenu.click();
+    await veterinariansMenuAll.click();
+    expect(await rafaelSpecialty.textContent()).toContain('dermatology')
+
+    await page.getByRole('link', { name: 'Specialties' }).click()
+    await rowSurgery.getByRole("button", { name: "Edit" }).click();
+    await editSpecialtyInput.click()
+    await editSpecialtyInput.clear()
+    await editSpecialtyInput.fill('surgery')
+    await page.getByRole('button', { name: "Update" }).click()
+})
+
+test('Validate specialty lists', async ({ page }) => {
+    await page.goto("/");
+    await page.getByRole('link', { name: 'Specialties' }).click()
+    await page.getByRole('button', { name: "Add" }).click()
+    await page.locator('#name').fill('oncology')
+    await page.getByRole('button', { name: "Save" }).click()
+
+    const allSpecialtiesValues = await page.locator('td input')
+    let specialtiesList = []
+
+    for (const specialty of await allSpecialtiesValues.all()) {
+        specialtiesList.push(await specialty.inputValue())
+    }
+
+    await page.getByText('Veterinarians').click()
+    await page.locator(".dropdown-menu").getByText(" All").click()
+    const sharonJenkinsRow = page.locator('tr', { has: page.getByText(' Sharon Jenkins ') })
+    await sharonJenkinsRow.getByRole('button', { name: 'Edit Vet' }).click()
+
+    await page.locator('.dropdown-display').click()
+    let specialtiesListFromDropDOwn = []
+    const allValuesFromDropDownMenu = await page.locator(".dropdown-content label").allTextContents()
+    specialtiesListFromDropDOwn.push(allValuesFromDropDownMenu)
+    expect(specialtiesList).toBe(specialtiesList)
```

Comment:
Something is wrong here. You are comparing the same objects, and of course, they are the same.
Also check the difference between toBe() and toEqual() in this blog post: https://www.bondaracademy.com/blog/tobe-vs-toequal-assertion-playwright

Reply:
Got it, thanks for the reference.